packages/application-generic/src/services/auth/auth.service.ts

```diff
@@ -124,6 +136,44 @@ export class AuthService {
     };
   }
 
+  private async updateUserUsername(
+    user: UserEntity,
+    profile: {
+      name: string;
+      login: string;
+      email: string;
+      avatar_url: string;
+      id: string;
+    },
+    authProvider: AuthProviderEnum
+  ) {
+    const withoutUsername = user.tokens.find(
+      (token) =>
+        token.provider === authProvider &&
+        !token.username &&
+        String(token.providerId) === String(profile.id)
+    );
+
+    if (withoutUsername) {
+      await this.userRepository.update(
+        {
+          _id: user._id,
+          'tokens.providerId': profile.id,
+        },
+        {
+          $set: {
+            'tokens.$.username': profile.login,
+          },
+        }
+      );
+
+      user = await this.userRepository.findById(user._id);
```

Comment:
This flow is a bit confusing even if I understand what it is doing.
I guess that the exception should be thrown if when updating the user is not found.
Also there must be an option to retrieve the updated entity in the return of the MongoDB functionality, saving us from doing the fetch later. I think we could review that part. Highly likely we would need to change our base repository class for that.